autogalaxy/analysis/analysis/dataset.py

```diff
@@ -0,0 +1,226 @@
+            return self._adapt_images
+
+        if self.adapt_image_maker is not None:
+            return self.adapt_image_maker.adapt_images
+
+    def set_preloads(self, paths: af.DirectoryPaths, model: af.Collection):
+        """
+        It is common for the model to have components whose parameters are all fixed, and thus the way that component
+        fits the data does not change. For example, if all parameter associated with the light profiles of galaxies
+        in the model are fixed, the image generated from these galaxies will not change irrespective of the model
+        parameters chosen by the non-linear search.
+
+        Preloading exploits this to speed up the log likelihood function, by inspecting the model and storing in memory
+        quantities that do not change. For the example above, the image of all galaxies would be stored in memory and
+        to perform every fit in the `log_likelihood_funtion`.
+
+        This function sets up all preload quantities, which are described fully in the `preloads` modules. This
+        occurs directly before the non-linear search begins, to ensure the model parameterization is fixed.
+
+        Parameters
+        ----------
+        paths
+            The PyAutoFit paths object which manages all paths, e.g. where the non-linear search outputs are stored,
+            visualization and the pickled objects used by the aggregator output by this function.
+        model
+            The PyAutoFit model object, which includes model components representing the galaxies that are fitted to
+            the imaging data.
+        """
+
+        logger.info(
+            "PRELOADS - Setting up preloads, may take a few minutes for fits using an inversion."
+        )
+
+        self.preloads = self.preloads_cls()
+
+        settings_inversion_original = copy.copy(self.settings_inversion)
+
+        self.settings_inversion.image_mesh_min_mesh_pixels_per_pixel = None
+        self.settings_inversion.image_mesh_adapt_background_percent_threshold = None
+
+        fit_maker = self.fit_maker_cls(model=model, fit_from=self.fit_from)
+
+        fit_0 = fit_maker.fit_via_model_from(unit_value=0.45)
+        fit_1 = fit_maker.fit_via_model_from(unit_value=0.55)
+
+        if fit_0 is None or fit_1 is None:
+            self.preloads = self.preloads_cls(failed=True)
+
+            self.settings_inversion = settings_inversion_original
+
+        else:
+            self.preloads = self.preloads_cls.setup_all_via_fits(
+                fit_0=fit_0, fit_1=fit_1
+            )
+
+            if conf.instance["general"]["test"]["check_preloads"]:
+                self.preloads.check_via_fit(fit=fit_0)
+
+            self.settings_inversion = settings_inversion_original
+
+        if isinstance(paths, af.DatabasePaths):
+            return
+
+        os.makedirs(paths.profile_path, exist_ok=True)
+        self.preloads.output_info_to_summary(file_path=paths.profile_path)

```

Comment:
Handle directory creation right before the data is output (e.g. in the `output_info_to_summary` method)